### PR TITLE
HHH-9907 - bytecode enhancer - fix resource path on windows

### PR DESCRIPTION
**Pull request by barreiro**

### CODE REVIEW

**Comment by dreab8** [comment]
thanks @barreiro

File: hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/Enhancer.java [hunk 1]
```diff
@@ -107,11 +105,8 @@ private ClassPool buildClassPool(EnhancementContext enhancementContext) {
 	}
 
 	private CtClass loadCtClassFromClass(ClassPool cp, Class<?> aClass) throws IOException {
-		return cp.makeClass( aClass.getClassLoader().getResourceAsStream( getFilenameForClass( aClass ) ) );
-	}
-
-	private String getFilenameForClass(Class<?> aClass) {
-		return aClass.getName().replace( '.', File.separatorChar ) + JavaFileObject.Kind.CLASS.extension;
+		String resourceName = aClass.getName().replace( '.', '/' ) + ".class";
+		return cp.makeClass( aClass.getClassLoader().getResourceAsStream( resourceName ) );
```
Comment (scottmarlow) [inline hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/Enhancer.java]: I wonder what closes the stream returned by getResourceAsStream()?  Should we close that?

Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/Enhancer.java]: I think we have to close it.

Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/Enhancer.java]: I'm going to fix it

**Pull request merged**